daemon/daemonbuilder/builder.go

```diff
@@ -174,11 +174,11 @@ func (d Docker) Copy(c *daemon.Container, destPath string, src builder.FileInfo,
 		}
 
 		// try to successfully untar the orig
-		if err := d.Archiver.UntarPath(srcPath, tarDest); err == nil {
-			return nil
-		} else if err != io.EOF {
-			logrus.Debugf("Couldn't untar to %s: %v", tarDest, err)
+		err := d.Archiver.UntarPath(srcPath, tarDest)
```

Comment:
Before it was being ignored, now it's not.

Previously, any error part way during extraction would fall through to adding the file. The only thing that `io.EOF` controlled was a log line that virtually nobody would see.
I've removed that conditional because an `io.EOF` (or any other error) bubbling up to this point means something *has* gone wrong during the untar (e.g. the file is too short to peek the magic bytes from) and so we should error.